modules/api/src/main/java/org/apache/ignite/compute/JobExecution.java

```diff
@@ -56,8 +56,8 @@ public interface JobExecution<R> {
      * Cancels the job.
      *
      * @return The future which will be completed with {@code true} when the job is cancelled, {@code false} when the job couldn't be
-     *         cancelled (either it's not yet started, or it's already completed), or {@code null} if the job no longer exists due to
-     *         exceeding the retention time limit.
+     *         cancelled (if it's not queued or executing), or {@code null} if the job no longer exists due to exceeding the retention time
```

Comment:
```suggestion
     *         cancelled (it is already completed), or {@code null} if the job no longer exists due to exceeding the retention time
```

Reply:
Let's say `it is already completed or cancelled` then?

Reply:
Or `if it's not queued or executing, meaning already completed or cancelled` to be very explicit.

Reply:
The last one looks good to me.